### DISCUSSION TITLE
Does phantom support compressed tokens?

### DISCUSSION THREAD

**Discussion by KonradPodstawski** [phantom/docs#291] [General]
With the introduction of zk-compression on mainnet, does Phantom already support zk-compression tokens? Will it be possible to decompress these tokens from within the wallet? (If I misunderstand how decompression works, I apologise).

**Comment by adamdelphantom** [comment]
Hey @KonradPodstawski,

We do not inherently support zk-compression, yet. According to the zk-compression [docs](https://www.zkcompression.com/introduction/intro-to-development#on-chain-program-development) the tokens should be compatible with the spl implementation but we haven't tested yet.

**Reply by KonradPodstawski** [reply]
Great, I'm waiting for it to be possible to decompress tokens from the wallet level it will be a game changer

**Reply by adamdelphantom** [reply]
I tested and it doesn't appear they show up. We don't have an eta yet for support but keep an eye on our socials for more updates.

**Reply by adamdelphantom** [reply]
@KonradPodstawski Do you have a specific use case in mind? If so would love to hear more about it to inform our priority of support.

**Reply by KonradPodstawski** [reply]
In my case, compressed tokens are ideal because they help reduce the costs of distribution to new users (we don't need to create token accounts for them, etc.). The issue is that we care a lot about UX, and we don't want the user to have to perform the decompression of tokens themselves (such as creating a token account) on our site or having to send SOL to cover transaction fees. We maintain a model where payouts are gasless, meaning there are no additional fees for the user.

From our perspective, the best system would be one where the user could decompress the token directly within their wallet (for example, by clicking on the compressed token). This would allow the user to automatically create a token account and decompress the tokens directly in their wallet without needing to go through extra steps on our site.
